automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PublishedWorkspaceSpec.scala

```diff
@@ -0,0 +1,184 @@
+package org.broadinstitute.dsde.test.api.orch
+
+import org.broadinstitute.dsde.test.LibraryData
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.UserPool
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.test.RandomUtil
+import org.broadinstitute.dsde.workbench.service.{Orchestration, Rawls}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{FreeSpec, Matchers}
+import org.scalatest.time.{Seconds, Span}
+import spray.json._
+
+
+class PublishedWorkspaceSpec extends FreeSpec with WorkspaceFixtures with BillingFixtures with Matchers with Eventually with RandomUtil {
+
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(30, Seconds)), interval = scaled(Span(30, Seconds)))
+
+  "For a user with publish permissions" - {
+    "a published workspace" - {
+
+      "should be visible in the library table" in {
+
+        val curatorUser = UserPool.chooseCurator
+        implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+        withCleanBillingProject(curatorUser) { billingProject =>
+          withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+            val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+            Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+            Orchestration.library.publishWorkspace(billingProject, workspaceName)
+
+            withClue("a published workspace should be visible in library.") {
+              eventually {
+                isVisibleInLibrary(workspaceName) shouldBe true
+              }
```

Comment:
Correct me if I'm but wrong, but I don't believe this requires an `eventually` - publishWorkspace returns after the workspace has been published, which means it should immediately be available to search.

Reply:
I saw few times where it took 2 calls to search library to get expected result.  `eventually` will help reduce test flakiness.